### PR TITLE
feat(search): Aggregator, step 1

### PR DESCRIPTION
**Pull request by dranikpg**
First step for aggregator for FT.AGGREGATE #2190. 

PS: Using FT.AGGREGATE will also allow sorting without sortable fields

### CODE REVIEW

**Comment by dranikpg** [comment]
Second step will be integrating this into the search pipeline itself, so don't worry that all the types are standalone

**Review by chakaz — DISMISSED** [review]

File: src/server/search/aggregator.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    return ValueIterator{};
+  }
+
+ private:
+  ValueIterator() = default;
+
+  std::string_view field_;
+  absl::Span<const DocValues> values_;
+
+  static const Value kEmpty;
```
[the remark this answers is not in the document]
Comment (dranikpg) [reply src/server/search/aggregator.h:61]: Then I have to move the impl...

**Review by chakaz — DISMISSED** [review]

File: src/server/search/aggregator.cc [hunk 2]
```diff
@@ -0,0 +1,126 @@
+#include "server/search/aggregator.h"
```
Comment (romange) [inline src/server/search/aggregator.cc:1]: please add the file header

File: src/server/search/aggregator_test.cc [hunk 3]
```diff
@@ -0,0 +1,102 @@
+#include "server/search/aggregator.h"
```
Comment (romange) [inline src/server/search/aggregator_test.cc:1]: file header

**Review by chakaz — APPROVED** [review]

**Pull request merged**